src/WWT.Providers/Providers/DSSProvider.cs

```diff
@@ -1,70 +1,58 @@
 using System;
-using System.Configuration;
 using System.IO;
 using WWTWebservices;
 
 namespace WWT.Providers
 {
     public class DSSProvider : RequestProvider
     {
-        public override void Run(WwtContext context)
+        private readonly IPlateTilePyramid _plateTile;
+
+        public DSSProvider(IPlateTilePyramid plateTile)
         {
-            string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
-            string dssTerapixelDir = ConfigurationManager.AppSettings["DssTerapixelDir"];
+            _plateTile = plateTile;
+        }
 
-            string query = context.Request.Params["Q"];
-            string[] values = query.Split(',');
-            int level = Convert.ToInt32(values[0]);
-            int tileX = Convert.ToInt32(values[1]);
-            int tileY = Convert.ToInt32(values[2]);
+        public override void Run(WwtContext context)
+        {
+            var (level, tileX, tileY) = context.GetLevelAndCoordinates();
 
-            int octsetlevel = level;
-            string filename;
+            using (var stream = GetStream(level, tileX, tileY))
+            {
+                if (stream is null)
+                {
+                    context.Response.Write("No image");
+                    context.Response.Close();
+                }
 
+                context.Response.ContentType = "image/png";
+                stream.CopyTo(context.Response.OutputStream);
+                context.Response.Flush();
+                context.Response.End();
+            }
+        }
 
+        private Stream GetStream(int level, int tileX, int tileY)
+        {
             if (level > 12)
             {
-                context.Response.Write("No image");
-                context.Response.Close();
-                return;
+                return null;
```

Comment:
This is one of those cases where I believe Jonathan said that is is important for the server to return a response that is not a PNG file, but not also not a 404 error. I'm not sure what returning `null` will do here, but I believe that it's important to preserve the existing behavior in this branch.

Reply:
Yes, this actually has the same behavior, just been refactored to be easier to follow. However, I'll switch it back to the previous implementation without so much refactoring since this change should focus on the file abstraction.